### PR TITLE
CI: Setup 3.9 Travis Build

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] closes https://github.com/pandas-dev/pandas/issues/33395

- In a very similar fashion to @jbrockmendel work for 3.8 - might be useful to have start a 3.9 build so we can start eliminating the issues @TomAugspurger mentioned in https://github.com/pandas-dev/pandas/issues/33395.

This is using currently using `Python 3.9.0a5+`

Currently failing with
```
ImportError while loading conftest '/home/travis/build/pandas-dev/pandas/pandas/conftest.py'.
pandas/conftest.py:1123: in <module>
    ("period", [pd.Period(2013), pd.NaT, pd.Period(2018)]),
pandas/_libs/tslibs/period.pyx:2470: in pandas._libs.tslibs.period.Period.__new__
    freq = Resolution.get_freq(reso)
pandas/_libs/tslibs/resolution.pyx:240: in pandas._libs.tslibs.resolution.Resolution.get_freq
    return cls._reso_freq_map[resostr]
E   AttributeError: type object 'type' has no attribute '_reso_freq_map'
The command "ci/run_tests.sh" exited with 4.
cache.2
store build cache
```

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Test failures currently related to: https://github.com/pandas-dev/pandas/issues/33507

**Comment by jbrockmendel** [comment]
@alimcmaster1 did we determine this needs to wait on an upstream fix?

File: ci/build39.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+python setup.py build_ext -inplace
+python -m pip install --no-build-isolation -e .
+
+python -c "import sys; print(sys.version_info)"
+python -c "import pandas as pd"
+python -c "import hypothesis"
+
+# TODO: Is there anything else in setup_env that we really want to do?
+# ci/setup_env.sh
```
Comment (jreback) [inline ci/build39.sh:26]: add a line at the end

**Review by jreback — CHANGES_REQUESTED** [review]
let us know when ready

**Comment by alimcmaster1** [comment]
> @alimcmaster1 did we determine this needs to wait on an upstream fix?

Yes waiting on a fix to the current cython alpha - https://github.com/cython/cython/issues/3544.

(We cant use current cython release due to a different issue - https://github.com/cython/cython/issues/3395)

Will ping on here when ready.

**Comment by TomAugspurger** [comment]
@alimcmaster1 you may need to manually skip the clipboard tests on this build.

**Comment by alimcmaster1** [comment]
Azure pipelines don't seem to be triggering

**Comment by alimcmaster1** [comment]
@pandas-dev/pandas-core 3.9 build is ready. (Azure seems to be playing up - all jobs have ran but doesn’t seem to be reporting back to my PR)

File: ci/build39.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -e
+# Special build for python3.9 until numpy puts its own wheels up
+
+sudo apt-get install build-essential gcc xvfb
+pip install --no-deps -U pip wheel setuptools
+pip install python-dateutil pytz pytest pytest-xdist hypothesis
+pip install cython --pre # https://github.com/cython/cython/issues/3395
+
+# Alternative for getting numpy:
+# pip install --pre -f https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com/ numpy
```
Comment (jreback) [inline ci/build39.sh:10]: do these wheels work yet?

**Review by jreback — CHANGES_REQUESTED** [review]
can u add an issue to revert this once conda builds are good 

a question about wheels

File: ci/build39.sh [hunk 3]
```diff
@@ -6,14 +6,7 @@ pip install --no-deps -U pip wheel setuptools
...
 
-# Alternative for getting numpy:
-# pip install --pre -f https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com/ numpy
-git clone https://github.com/numpy/numpy
-cd numpy
-python setup.py build_ext --inplace
-python setup.py install
-cd ..
-rm -rf numpy
+pip install --pre -f https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com numpy
```
Comment (jreback) [inline ci/build39.sh:9]: this might not be the right address iirc this changed
maybe have to look on the numpy issue list

Comment (alimcmaster1) [reply ci/build39.sh:9]: Its the same address we use in the numpy dev build? - https://github.com/pandas-dev/pandas/blob/master/ci/deps/azure-37-numpydev.yaml#L19

Comment (alimcmaster1) [reply ci/build39.sh:9]: Its the same address we use in the numpy dev build? - https://github.com/pandas-dev/pandas/blob/master/ci/deps/azure-37-numpydev.yaml#L19

**Comment by alimcmaster1** [comment]
all green & raised an issue for later follow ups @jreback

File: ci/build39.sh (hunk 3 above)
Comment (jreback) [reply ci/build39.sh:9]: these might be older: 


https://anaconda.org/scipy-wheels-nightly/numpy/files
i think is the new location (includes scipy as well)

let’s update

Comment (alimcmaster1) [reply ci/build39.sh]: Ahh nice - thanks wasn't aware! Will update the numpy dev build too in a separate PR.

thanks!

**Comment by alimcmaster1** [comment]
Resorted to building numpy from source:

* https://anaconda.org/scipy-wheels-nightly/numpy/files has no 3.9 whl available yet.
* previous attempting running pip install --pre -f https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com numpy`
was just grabbing latest numpy release - whereas we probably want master/nightly build here :

```
Collecting numpy
  Downloading numpy-1.18.3.zip (5.4 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Building wheels for collected packages: numpy
  Building wheel for numpy (PEP 517): started
  Building wheel for numpy (PEP 517): still running...
  Building wheel for numpy (PEP 517): still running...
  Building wheel for numpy (PEP 517): finished with status 'done'
  Created wheel for numpy: filename=numpy-1.18.3-cp39-cp39-linux_x86_64.whl size=15218460 sha256=a8ece5537cef5c3f6110576f4937c80bfc60d6a3d4a6c6d27353b2e3dc6f677e
  Stored in directory: /home/travis/.cache/pip/wheels/b0/63/be/cd7680896e3713ce13b5b5748563e78e0cfe5ea9c0c20143a1```

**Comment by alimcmaster1** [comment]
@jreback @jbrockmendel any comments here? All green

File: ci/build39.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash -e
+# Special build for python3.9 until numpy puts its own wheels up
+
+sudo apt-get install build-essential gcc xvfb
+pip install --no-deps -U pip wheel setuptools
+pip install python-dateutil pytz pytest pytest-xdist hypothesis
+pip install cython --pre # https://github.com/cython/cython/issues/3395
+
+git clone https://github.com/numpy/numpy
+cd numpy
```
Comment (jreback) [inline ci/build39.sh:10]: can we use the wheel instead?
iirc the existing dev build was already changed

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/build39.sh (hunk 4 above)
Comment (alimcmaster1) [reply ci/build39.sh:10]: I tried above - but there are no 3.9 wheels available yet on scipy-wheels-night or the old rackcdn url

Comment (jreback) [reply ci/build39.sh:10]: hmm ok

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @alimcmaster1

File: ci/build39.sh (hunk 4 above)
Comment (alimcmaster1) [reply ci/build39.sh:10]: Will update when available - tracking in https://github.com/pandas-dev/pandas/issues/33948

**Pull request merged**